src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -173,23 +174,26 @@ public void LookForModuleLevelAndGlobalSuppressions (ModuleDefinition module, IC
 				if (!TryDecodeSuppressMessageAttributeData (instance, out info))
 					continue;
 
-				if (info.Target == null && (info.Scope == "module" || info.Scope == null)) {
+				// If the scope is missing we treat the suppression as if it was placed on the module.
+				var scope = info.Scope?.ToLower ();
+				if (info.Target == null && (scope == "module" || scope == null)) {
 					AddSuppression (info, provider);
 					continue;
 				}
 
-				if (info.Target == null || info.Scope == null)
+				if (info.Target == null || scope == null)
 					continue;
 
-				switch (info.Scope) {
+				switch (scope) {
 				case "type":
 				case "member":
 					foreach (var result in DocumentationSignatureParser.GetMembersForDocumentationSignature (info.Target, module))
```

Comment:
Should we be validating that the scope kind matches the target, and warning if you for example use scope "type" but specify a member signature as the target?

I also think we should make it a warning if the target doesn't match anything.

Reply:
Feels a little bit as an overkill since we'll end up running the DocumentationSignatureParse, but we can add the check. Related to this, I opened https://github.com/mono/linker/issues/2130.